04-cond.Rmd

```diff
@@ -212,6 +212,38 @@ In this case, "either" means "either or both", not "either one or the other but
 > plot_dist(dat[1:5, 10], threshold = 10)                    # samples (rows) 1-5 on day (column) 10
 > ```
 
+
+> ## Challenge - Putting together data analysis, conditionals, and plotting   {.challenge}
+>   + For this challenge, fill in the blanks of the function hist_max() below to create a function that does the following:  
+  
+> ```{r, eval=FALSE}
+1.For each patient (row) in a given file, find the max inflammation value and store these in a variable called 'max_patient_inflammation'.  
+2.Plot the max values into a simple histogram using the function hist() if and only if the spread (difference between largest and smallest) of the max values for all the patients is greater than your desired spread value, otherwise, return the values.  
+```
+
+> ``` 
+> hist_max <- function(dat, spread) {
+>  max_patient_inflammation <- ______(dat, __, max)
+>  difference <- ___(max_patient_inflammation) - ___(max_patient _inflammation)
+>  if (______ > ______) {
+>      hist(max_patient_inflammation)
+>  } else {
+>      return(max_patient_inflammation)
+>  }
+>}
+>```
+
+> ``` {r, include=FALSE}
+> hist_max <- function(dat, spread) {
+>  max_patient_inflammation <- apply(dat, 1, max)
+>  difference <- max(max_patient_inflammation) - min(max_patient_inflammation)
+>  if (difference > spread) {
+>      hist(max_patient_inflammation)
+>  } else {
```

Comment:
I am a little confused here by the use case. Why would the function only return the data if the histogram is not made? Shouldn't it return it either way? For example, this makes more sense to me.

```
if (difference > spread) {
  hist(max_patient_inflammation)
}
return(max_patient_inflammation)
```